python-client/giskard/scanner/prediction/metric.py

```diff
@@ -0,0 +1,143 @@
+from giskard.models.base import BaseModel
+from giskard.datasets.base import Dataset
+import pandas as pd
+from giskard.models._precooked import PrecookedModel
+
+
+class OverconfidenceDetector:
+    model = None
+    dataset = None
+    raw, prediction, raw_prediction, probabilities, all_predictions = None, None, None, None, None
+    ground_truth = None
+    all_proba_with_true_label_proba = None
+    cost = None
+
+    def __init__(self, model: BaseModel, dataset: Dataset):
+        self.model = model
+        self.dataset = dataset.copy()
+        if isinstance(model, PrecookedModel):
+            self.all_predictions = model._predictions.all_predictions
+        else:
+            raw, prediction, raw_prediction, probabilities, all_predictions = model.predict(
+                self.dataset)
+            self.all_predictions = all_predictions[1]
+        self.ground_truth = pd.DataFrame(dataset.df[dataset.target])
+        self._get_true_label_proba()
+        self._detect_issues()
```

Comment:
This stuff should not be done on `__init__`! Also you are not implementing the correct `Detector` interface:
```py
class Detector(ABC):
    @abstractmethod
    def run(self, model: BaseModel, dataset: Dataset) -> Sequence[Issue]:
        ...
 ```